includes/class-newspack-ads-settings.php

```diff
@@ -0,0 +1,374 @@
+<?php
+/**
+ * Newspack Ads Settings
+ *
+ * @package Newspack
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Newspack Ads Settings Class.
+ */
+class Newspack_Ads_Settings {
+
+	const API_NAMESPACE      = 'newspack-ads/v1';
+	const API_CAPABILITY     = 'manage_options';
+	const OPTION_NAME_PREFIX = '_newspack_ads_';
+
+	/**
+	 * Initialize settings.
+	 */
+	public static function init() {
+		add_action( 'rest_api_init', [ __CLASS__, 'register_api_endpoints' ] );
+	}
+
+	/**
+	 * Register the endpoints needed to fetch and update settings.
+	 */
+	public static function register_api_endpoints() {
+
+		register_rest_route(
+			self::API_NAMESPACE,
+			'/settings-list',
+			[
+				'methods'             => \WP_REST_Server::READABLE,
+				'callback'            => [ __CLASS__, 'api_get_settings_list' ],
+				'permission_callback' => [ __CLASS__, 'api_permissions_check' ],
+			]
+		);
+
+		register_rest_route(
+			self::API_NAMESPACE,
+			'/settings',
+			[
+				'methods'             => \WP_REST_Server::READABLE,
```

Comment:
I find it confusing that there are two ways of `GET`ting the settings list – one with section data (`/settings-list`), other "raw" (`/settings`). Also only the former is used in the proposed settings consumer (https://github.com/Automattic/newspack-plugin/pull/1208), which leaves the latter without a use case. 
I'd suggest removing `/settings` route and the associated code, and renaming `settings-list` to `settings`. This would make for a clearer API with a `/settings` endpoint that handles `GET`s and `POST`s.

Reply:
Good call! My thoughts were to reproduce the methods logic, which is for the `get_settings_list()` -> `/settings-list` to return an array with metadata for UI and the regular `get_settings()` -> `/settings` to return raw data object.

Since we don't have a use case for the initially proposed `/settings` endpoint, I agree we should simplify this and later revisit if its the case for a different approach.

Done in https://github.com/Automattic/newspack-ads/pull/220